nselib/httpcookies.lua

```diff
@@ -0,0 +1,476 @@
+--   local cookiejar = httpcookies.CookieJar:new(cookie_table)
+--   cookiejar:set_no_cookie_overwrite(true)
+--   
+--   local response
+--   response = cookiejar:get(host, port, path, options)
+--
+--   return response
+-- </code>
+--
+-- The following sample command tells us how to make use of the library
+-- in the existing scripts :
+-- 
+-- $nmap --script http-xss-scanner --script-args httpcookies.cookiejar={<table of cookies>}
+-- 
+-- Using the above command, the http library will parse httpcookies arguments 
+-- and makes an object of cookiejar class. This will help us to browse through
+-- the pages with session cookies. Besides, it also merges the newly received 
+-- cookies with the existing cookie jar.
+--
+-- The library supports an options table which provides the following functionlity
+-- * <code>no_cookie_overwrite</code>: With this option set as true, if we receive another
+-- cookie with the same name and attributes, the previous cookie wont be updated. (default:true)
+--
+--
+-- @author Paulino Calderon <calderon@websec.mx>
+-- @author Vinamra Bhatia 
+--
+---
+
+local io = require "io"
+local nmap = require "nmap"
+local stdnse = require "stdnse"
+local string = require "string"
+local table = require "table"
+local os = require "os"
+local shortport  = require "shortport"
+local http = require "http"
+local unittest = require "unittest"
+
+_ENV = stdnse.module("httpcookies", stdnse.seeall)
+
+local LIBRARY_NAME = "httpcookies"
+
+--TODO:
+--Incorporate more options in the library.
+
+-- The Cookies Class
+CookieJar = {
+
+  -- creates a new instance of CookieJar
+  -- @param cookies A table or string containing cookies 
+  -- @param options A table containing various options for the library.
+  -- @return o new instance of CookieJar
+  new = function(self, cookies, options)
+    local o = {
+      cookies = cookies or {},
+      options = options or {},
+    }
+
+    setmetatable(o, self)
+    self.__index = self
+
+    --Setting default values for the options
+    o.options.no_cookie_overwrite = o.options.no_cookie_overwrite or true
+
+    if ( o:parse(self.cookies) ) then
+      return o
+    end
+
+  end,
+
+
+  --- Parses the cookie and and splits it into its attributes if its a string.
+  -- @param cookies A cookie table 
+  -- @return status true on success, false on failure
+  parse = function(cookies)
+    if cookies == nil then
+      return true
+    end
+    if (type(cookies) == 'table') then
+      --Name and Value must be present in the cookie, rest
+      --attributes are optional.
+      --So, if we have a table of cookies, we take the first table 
+      --and then loop for all the attributes in that table.
+      for _,cookie in ipairs(cookies) do
+        for cookie_attribute, cookie_attribute_value in pairs(cookie) do
+          cookie_attribute = cookie_attribute:lower()
+          if(cookie_attribute == 'name') then 
+          if(type(cookie_attribute_value) ~= 'string') then
```

Comment:
Please fix all indentation issues. Remember to follow the code guidelines of the project.